### PR TITLE
KAFKA-14404: fix overlap of streams-config sections & describe additional parameters

### PR DESCRIPTION
**Pull request by AyoubOm**
*There is an overlap between two sections in the current version of streams-config.html as described in KAFKA-14404.
The goal of this fix is to remove the overlap as well as describing some other parameters and their default values*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by AyoubOm** [comment]
@ableegoldman Would you have time to check this ? If so, please let me know if you think we should describe any other fields here. Thanks :)

File: docs/streams/developer-guide/config-streams.html [hunk 1]
```diff
@@ -409,12 +409,12 @@ <h4><a class="toc-backref" href="#id23">num.standby.replicas</a><a class="header
...
-            <td>-1</td>
+            <td><code class="docutils literal"><span class="pre">-1</span></code></td>
           </tr>
           <tr class="row-even"><td>retry.backoff.ms</td>
-              <td>Medium</td>
-              <td colspan="2">The amount of time in milliseconds, before a request is retried. This applies if the <code class="docutils literal"><span class="pre">retries</span></code> parameter is configured to be greater than 0. </td>
-              <td>100 milliseconds</td>
+            <td>Medium</td>
+            <td colspan="2">The amount of time in milliseconds, before a request is retried. This applies if the <code class="docutils literal"><span class="pre">retries</span></code> parameter is configured to be greater than 0. </td>
+            <td>100 milliseconds</td>
```
Comment (mjsax) [inline docs/streams/developer-guide/config-streams.html]:
```suggestion
            <td><code class="docutils literal"><span class="pre">100</span></code></td>
```

Comment (mjsax) [reply docs/streams/developer-guide/config-streams.html]: The description already say:
> The amount of time in milliseconds

File: docs/streams/developer-guide/config-streams.html [hunk 2]
```diff
@@ -1111,28 +1111,31 @@ <h4><a class="toc-backref" href="#id18">Default Values</a><a class="headerlink"
...
+        <tr class="row-odd"><td>linger.ms</td>
           <td>Producer</td>
           <td>100</td>
         </tr>
         <tr class="row-even"><td>max.poll.records</td>
           <td>Consumer</td>
           <td>1000</td>
         </tr>
+        <tr class="row-odd">
+          <td>transaction.timeout.ms</td>
```
Comment (mjsax) [inline docs/streams/developer-guide/config-streams.html]: We set this only if EOS is enabled. Might be worth to call out?

We also set `delivery.timeout.ms` to "max integer" and enable idempotency if EOS is enabled.

File: docs/streams/developer-guide/config-streams.html [hunk 3]
```diff
@@ -1111,28 +1111,31 @@ <h4><a class="toc-backref" href="#id18">Default Values</a><a class="headerlink"
...
+          <td>Producer</td>
+          <td><code class="docutils literal"><span class="pre">10000</span></code></td>
+        </tr>
+        <tr class="row-even">
+          <td>client.id</td>
+          <td>-</td>
+          <td><code class="docutils literal"><span class="pre">&lt;application.id&gt;-&lt;random-UUID&gt;</span></code></td>
```
Comment (mjsax) [inline docs/streams/developer-guide/config-streams.html:1125]: That not correct. And it's more complicated overall...

By default, ie, if `client.id` is not provided by the user, we set an internal `clientId = applicationId + "-" + processId` (with processId being a UUID), ie, what you describe.

However, for different clients (consumer/producer/admin) we generate more complex ones:
- main consumer (`client.id` per thread):
  `clientId + "-StreamThread-" + threadIdx + "-consumer"`
- restore consumer (`client.id` per thread):
  `clientId + "-StreamThread-" + threadIdx + "-restore-consumer"`
- global consumer:
  `clientId + "-global-consumer"`
- producer (alos + eos-beta; `client.id` per thread):
  `clientId + "-StreamThread-" + threadIdx + "-producer"`
- producer (eos-alpha; `client.id` per task):
  `clientId + "-StreamThread-" + threadIdx + "-" + taskId + "-producer"`
- admin:
  `clientId + "-admin"`

File: docs/streams/developer-guide/config-streams.html [hunk 4]
```diff
@@ -1111,28 +1111,31 @@ <h4><a class="toc-backref" href="#id18">Default Values</a><a class="headerlink"
           <td>Consumer</td>
```
Comment (mjsax) [inline docs/streams/developer-guide/config-streams.html:1111]: Just saw the following above:
```
For detailed descriptions
        of these configs, see <a class="reference external" href="http://kafka.apache.org/0100/documentation.html#producerconfigs">Producer Configs</a>
        and <a class="reference external" href="http://kafka.apache.org/0100/documentation.html#newconsumerconfigs">Consumer Configs</a>.</p>
```
Still links to 0.10.0... Can we fix it?

Comment (mjsax) [reply docs/streams/developer-guide/config-streams.html:1111]: We also set `max.poll.records` to 1000.

File: docs/streams/developer-guide/config-streams.html [hunk 5]
```diff
@@ -1111,28 +1111,31 @@ <h4><a class="toc-backref" href="#id18">Default Values</a><a class="headerlink"
...
-        <code class="docutils literal"><span class="pre">allow.auto.create.topics</span></code>, your value
-        is ignored and setting it has no effect in a Kafka Streams application. You can set the other parameters.
-        Kafka Streams sets them to different default values than a plain
-        <code class="docutils literal"><span class="pre">KafkaConsumer</span></code>.
-      <p>Kafka Streams uses the <code class="docutils literal"><span class="pre">client.id</span></code>
-        parameter to compute derived client IDs for internal clients. If you don't set
-        <code class="docutils literal"><span class="pre">client.id</span></code>, Kafka Streams sets it to
-        <code class="docutils literal"><span class="pre">&lt;application.id&gt;-&lt;random-UUID&gt;</span></code>.
+      <p>Some parameters are not configurable by the user. If you supply a value that is different from the default value,
+        your value is ignored. Below is a list of some of these parameters.
```
Comment (mjsax) [inline docs/streams/developer-guide/config-streams.html]: We also set consumer `isolation.level` to `READ_COMMITTED` if EOS is enabled.

**Review by mjsax — COMMENTED** [review]
Thanks for the PR. Couple of comments.

File: docs/streams/developer-guide/config-streams.html (hunk 4 above)
Comment (AyoubOm) [reply docs/streams/developer-guide/config-streams.html:1111]: Version fixed. For `max.poll.records` it was added in the previous commit

File: docs/streams/developer-guide/config-streams.html (hunk 5 above)
Comment (AyoubOm) [reply docs/streams/developer-guide/config-streams.html]: Added `isolation.level` and `enable.idempotence` in a separate table. Is it fine ?

File: docs/streams/developer-guide/config-streams.html (hunk 2 above)
Comment (AyoubOm) [reply docs/streams/developer-guide/config-streams.html]: Added the parameters in a separate table. What do you think ?

**Comment by AyoubOm** [comment]
> Thanks for the PR. Couple of comments.

Thanks @mjsax for the review. Please check the changes pushed

File: docs/streams/developer-guide/config-streams.html (hunk 3 above)
Comment (AyoubOm) [reply docs/streams/developer-guide/config-streams.html:1125]: I see, would it be more convenient to not describe it in this case ?

File: docs/streams/developer-guide/config-streams.html (hunk 1 above)
Comment (AyoubOm) [reply docs/streams/developer-guide/config-streams.html]: Fixed, thanks !

File: docs/streams/developer-guide/config-streams.html [hunk 6]
```diff
@@ -1097,8 +1097,8 @@ <h4><a class="toc-backref" href="#id17">Naming</a><a class="headerlink" href="#n
     <div class="section" id="default-values">
       <h4><a class="toc-backref" href="#id18">Default Values</a><a class="headerlink" href="#default-values" title="Permalink to this headline"></a></h4>
       <p>Kafka Streams uses different default values for some of the underlying client configs, which are summarized below. For detailed descriptions
-        of these configs, see <a class="reference external" href="http://kafka.apache.org/0100/documentation.html#producerconfigs">Producer Configs</a>
-        and <a class="reference external" href="http://kafka.apache.org/0100/documentation.html#newconsumerconfigs">Consumer Configs</a>.</p>
+        of these configs, see <a class="reference external" href="http://kafka.apache.org/documentation.html#producerconfigs">Producer Configs</a>
+        and <a class="reference external" href="http://kafka.apache.org/documentation.html#newconsumerconfigs">Consumer Configs</a>.</p>
```
Comment (ableegoldman) [inline docs/streams/developer-guide/config-streams.html]:
```suggestion
        of these configs, see <a class="reference external" href="https://kafka.apache.org/documentation.html#producerconfigs">Producer Configs</a>
        and <a class="reference external" href="https://kafka.apache.org/documentation.html#consumerconfigs">Consumer Configs</a>.</p>
```

File: docs/streams/developer-guide/config-streams.html (hunk 3 above)
Comment (ableegoldman) [reply docs/streams/developer-guide/config-streams.html:1125]: People do ask about the `cliend.id` fairly often, so imho it's a common enough source of confusion that it's worth laying out in the docs, even if it's a bit complicated. Maybe it doesn't all fit well into this table but perhaps you can add a short section about the `client.id` somewhere above/below this table, and then for the "default" value just write "See here for details" with a link to that section...something like that?

File: docs/streams/developer-guide/config-streams.html (hunk 6 above)
Comment (AyoubOm) [reply docs/streams/developer-guide/config-streams.html]: fixed

**Comment by AyoubOm** [comment]
@ableegoldman @mjsax I added a section for `client.id`, the result looks like this:

Capture d’écran 2024-01-19 à 15 59 39

Please let me know if you have any other suggestion.

**Review by ableegoldman — APPROVED** [review]
LGTM -- thanks for the improvement!

**Comment by ableegoldman** [comment]
Merged to trunk. Thanks for the PR @AyoubOm -- one quick tip, make sure to format the PR title prefix as "KAFKA-14404" rather than [Kafka-14404]. Just good to know if you ever submit another PR (and we hope you will!)

**Comment by mjsax** [comment]
Thanks for pushing this over the finish line @AyoubOm! Great PR.

**Pull request merged**